exchanges/bitget/bitget.go

```diff
@@ -0,0 +1,4840 @@
+	errStrategyMutex                  = errors.New("only one of immediate or cancel, fill or kill, and post only can be set to true")
+	errAuthenticatedWebsocketDisabled = errors.New("authenticatedWebsocketAPISupport not enabled")
+	errAssetModeEmpty                 = errors.New("assetMode cannot be empty")
+	errTakeProfitTriggerPriceEmpty    = fmt.Errorf("%w: TakeProfitTriggerPrice below minimum", limits.ErrPriceBelowMin)
+	errStopLossTriggerPriceEmpty      = fmt.Errorf("%w: StopLossTriggerPrice below minimum", limits.ErrPriceBelowMin)
+	errOrderIDMutex                   = errors.New("exactly one of orderID and clientOrderID must be set")
+	errProductTypeAndPairEmpty        = errors.New("productType and pair cannot both be empty")
+
+	prodTypes = []string{"USDT-FUTURES", "COIN-FUTURES", "USDC-FUTURES"}
+	planTypes = []string{"normal_plan", "track_plan", "profit_loss"}
+)
+
+// QueryAnnouncements returns announcements from the exchange, filtered by type and time
+func (e *Exchange) QueryAnnouncements(ctx context.Context, annType string, startTime, endTime time.Time, pagination uint64, limit uint8) ([]AnnouncementResp, error) {
+	var params Params
+	params.Values = make(url.Values)
+	if err := params.prepareDateString(startTime, endTime, true, true); err != nil {
+		return nil, err
+	}
+	params.Values.Set("annType", annType)
+	params.Values.Set("language", "en_US")
+	params.Values.Set("idLessThan", strconv.FormatUint(pagination, 10))
+	if limit > 0 {
+		params.Values.Set("limit", strconv.FormatUint(uint64(limit), 10))
+	}
+	var resp []AnnouncementResp
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, rate20, bitgetPublic+bitgetAnnouncements, params.Values, &resp)
+}
+
+// GetTime returns the server's time
+func (e *Exchange) GetTime(ctx context.Context) (*TimeResp, error) {
+	var resp *TimeResp
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, rate20, bitgetPublic+bitgetTime, nil, &resp)
+}
+
+// GetTradeRate returns the fees the user would face for trading a given symbol
+func (e *Exchange) GetTradeRate(ctx context.Context, pair currency.Pair, businessType string) (*TradeRateResp, error) {
+	if pair.IsEmpty() {
+		return nil, currency.ErrCurrencyPairEmpty
+	}
+	if businessType == "" {
+		return nil, errBusinessTypeEmpty
+	}
+	vals := url.Values{}
+	vals.Set("symbol", pair.String())
+	vals.Set("businessType", businessType)
+	var resp *TradeRateResp
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, rate10, http.MethodGet, bitgetCommon+bitgetTradeRate, vals, nil, &resp)
+}
+
+// GetAllTradeRates returns the fees the user would face for all symbols within a given business type
+func (e *Exchange) GetAllTradeRates(ctx context.Context, businessType string) ([]AllTradeRatesResp, error) {
+	if businessType == "" {
+		return nil, errBusinessTypeEmpty
+	}
+	vals := url.Values{}
+	vals.Set("businessType", businessType)
+	var resp []AllTradeRatesResp
```

Comment:
can use []* for all response type declarations from now on; for consistency.
Instead of iterating over a slice and use the index to access elements, we can iterate over the slice but use the pointer and reduce the memory overhead.